server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java

```diff
@@ -420,7 +422,20 @@ private void assertNodesRemovedAfterZoneDecommission(boolean originalClusterMana
             tempZones.remove(originalClusterManagerZone);
             zoneToDecommission = randomFrom(tempZones);
         }
-        String activeNode = getNonDecommissionedNode(internalCluster().clusterService().state(), zoneToDecommission);
+        String activeNode;
+        switch (zoneToDecommission) {
```

Comment:
If you want to make this cleaner, make a map of zones -> nodes. It will avoid having 4 variables of `nodes_in_zone_xyz` and you could directly use the map here to pick the active node.

Reply:
Makes sense. Refactored a bit